Use dep package manager

sevoma opened the pull request:
I didn't see anything managing dependencies, so I'm introducing dep package manager.

https://github.com/golang/dep

The usage is `dep ensure` to grab all of the dependencies in the `Gopkg.lock` file and place them into the vendor folder.
https://github.com/golang/dep#usage

crewjam commented on Gopkg.lock:
```diff
@@ -0,0 +1,75 @@
+  version = "v1.0.0"
+
+[[projects]]
+  name = "github.com/crewjam/saml"
```
Other than watching a gophercon talk about it, I've not used `dep`, so forgive a bit of ignorance here. Is it right to have our own package in the list?

sevoma replied:
Good catch, that happened because I was running it from github.com/sevoma/saml folder. When run from crewjam/saml, it does not appear and works properly.

sevoma commented:
I'm not sure how the existing travis builds pull in dependencies, but that's a missing piece of this. Travis should run `dep ensure` to pull in the proper dependency versions.

Another important thing to note:
There are many dependency managers to choose from. `dep` is likely to become the official golang package manager, and works quite nicely, but the `dep` tool itself only compiles with golang 1.8. You can use the tool on golang 1.7 projects though.

crewjam commented:
Good point about travis. Looks like it might just magically work: https://docs.travis-ci.com/user/languages/go/#godep-support

sevoma commented:
Apologies for the build spam if you've been receiving that.
I updated .travis.yml to use `dep` for building, and added go 1.9 to the build list.

I think everything is in place now if you'd like to merge.
